### DISCUSSION TITLE
Connect to WiFi and esp8266

### DISCUSSION THREAD

**Discussion by jensahylda** [SmingHub/Sming#2579] [Ideas]
Hello

I have a feeling that when i connect to a WiFi with multiple identical networks. Say a WiFi router with name "MyRouter" and password "12345678". And then an extender with same name "MyRouter" and password "12345678".

Then its random which one is connected to. I would expect the SSID with the best signal quality is chosen but i think its random.

Anyone knows if this assumption is correct? If yes, any hints where to modify Sming to achieve this?

Regards Jens

**Comment by mikee47** [comment]
See #2522 which added an overload to `Station::config()` which takes a `StationClass::Config` structure. Here's a brief how-to:

```
Config cfg{
	.ssid = WIFI_SSID,
 	.password = WIFI_PASSWORD,
 	.bssid = "Actual BSSID",
	.autoConnectOnStartup = ...,
	.save = ...,
};
WifiStation,config(cfg);
```

You'll need to find the station BSSID and pass it in the config call.

**Comment by jensahylda** [comment]
Thank you very much.

fre. 4. nov. 2022 17.23 skrev Mike ***@***.***>:
